app/core/epoch_sealing_job.py

```
@@ -0,0 +1,195 @@
+                                        await epoch_repo.mark_epoch_failed(epoch.id)
+                                    else:
+                                        logger.exception(
+                                            "Failed to seal epoch %s (attempt %s); will retry",
+                                            epoch.id,
+                                            new_count,
+                                        )
+                                except Exception as persist_err:
+                                    logger.exception(
+                                        "Failed to persist seal retry count or mark epoch FAILED for %s: %s",
+                                        epoch.id,
+                                        persist_err,
```

Ruff (0.15.5)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)